docs/Agent/user-guide/man_page.md

```diff
@@ -1 +1,485 @@
+neither value is available, the result of `hostname -f` is used.
+(If no value is available, the command will exit with an error.)
+
+`--prefix`  
+This option allows the user to specify an optional
+directory-path hierarchy to be used when displaying the result
+tar balls on the pbench server.
+
+`--show-server`  
+This will not move any results but will resolve and
+then display the pbench server destination for results.
+
+`--xz-single-threaded`  
+This will force the use of a single
+thread for locally compressing the result files.
+
+`--help`  
+Show this message and exit.
+
+---
+
+### pbench-register-tool
+
+---
+
+**NAME**  
+`pbench-register-tool` - registers the specified tool
+
+**SYNOPSIS**  
+`pbench-register-tool [OPTIONS]`
+
+**DESCRIPTION**  
+Register the specified tool.  
+List of available tools:
+
+**Transient**
+
+- blktrace
+- bpftrace
+- cpuacct
+- disk
+- dm-cache
+- docker
+- docker-info
+- external-data-source
+- haproxy-ocp
+- iostat
+- jmap
+- jstack
+- kvm-spinlock
+- kvmstat
+- kvmtrace
+- lockstat
+- mpstat
+- numastat
+- oc
+- openvswitch
+- pcp-transient
+- perf
+- pidstat
+- pprof
+- proc-interrupts
+- proc-sched_debug
+- proc-vmstat
+- prometheus-metrics
+- qemu-migrate
+- rabbit
+- sar
+- strace
+- sysfs
+- systemtap
+- tcpdump
+- turbostat
+- user-tool
+- virsh-migrate
+- vmstat
+
+**Persistent**
+
+- node-exporter
+- dcgm
+- pcp
+
+For a list of tool-specific options, run:
+
+> `/opt/pbench-agent/tool-scripts/<tool-name> --help`
+
+**OPTIONS**  
+`--name` STR  
+<`STR`> specifies the name of the tool to be registered.
+
+`-g`, `--group`, `--groups` STR  
+Register the tool in <`STR`> group. If no group is specified, the `default` group is assumed.
+
+`[--persistent | --transient]`  
+For tools which can be run as either "transient" (where they are started and stopped on each iteration) or as "persistent" (where they are started before the first iteration and run continuously over all iterations), these options determine how the tool will be run. (Most tools are configured to run as either or the other, so these options are not necessary in those cases, and specifying the wrong one will produce an error.)
+
+`--remotes`  
+Single remote host, a list of remote hosts (comma-separated, no spaces) or an "at" sign (`@`) followed by a filename. In this last case, the file should contain a list of hosts and their (optional) labels. Each line of the file should contain a hostname, optionally followed by a label separated by a comma (`,`); empty lines are ignored, and comments are denoted by a leading hash (`#`), character.
+
+`--help`  
+Show this message and exit.
+
+---
+
+### pbench-register-tool-set
+
+---
+
+**NAME**  
+`pbench-register-tool-set` - register the specified toolset
+
+**SYNOPSIS**  
+`pbench-register-tool-set [OPTIONS]`
+
+**DESCRIPTION**  
+Register all the tools in the specified toolset.
+
+Available <`tool-set`> from /opt/pbench-agent/config/pbench-agent.cfg:
```

Comment:
This should probably be `<tool-set>` for consistency with the description below...